cistem/protocols/protocol_tomo_resample.py

```diff
@@ -0,0 +1,184 @@
+
+    More info:
+        https://cistem.org
+    '''
+
+    _label = 'resample tomogram'
+    _possibleOutputs = {OUTPUT_TOMO_NAME: SetOfTomograms}
+    _devStatus = BETA
+
+    tomoList = []
+
+    def __init__(self, **args):
+        EMProtocol.__init__(self, **args)
+
+    # -------------------------- DEFINE param functions ----------------------
+    def _defineParams(self, form):
+        ''' Define the input parameters that will be used.
+        Params:
+            form: this is the form to be populated with sections and params.
+        '''
+        #     Example 'resample' input:
+        # 
+        #     resample
+        #  
+        #  
+        #         **   Welcome to Resample   **
+        #  
+        #              Version : 1.00
+        #             Compiled : Dec  2 2017
+        #                 Mode : Interactive
+        #  
+        # Input image file name [tomo2_L1G1_72_d3.rec]       : tomo2_L1G1-dose_filt.rec
+        # Output image file name [tomo2_L1G1_72_d3-bin8.rec] : tomo2_L1G1_72_d3-bin8.rec
+        # Is the input a volume [YES]                        : YES
+        # New X-Size [464]                                   : 464
+        # New Y-Size [464]                                   : 464
+        # New Z-Size [232]                                   : 232
+
+
+        # You need a params to belong to a section:
+        form.addSection(label=Message.LABEL_INPUT)
+        form.addParam('inTomograms', PointerParam,
+                        pointerClass='SetOfTomograms',
+                        allowsNull=False,
+                        label='Input tomograms')
+
+        form.addParam('newXsize', IntParam,
+                        default=512,
+                        label='New X-Size',
+                        help='Volume will be rescaled to this size in X dimension (voxels)')
+        form.addParam('newYsize', IntParam,
+                        default=512,
+                        label='New Y-Size',
+                        help='Volume will be rescaled to this size in Y dimension (voxels)')
+        form.addParam('newZsize', IntParam,
+                        default=256,
+                        label='New Z-Size',
+                        help='Volume will be rescaled to this size in Z dimension (voxels)')
+
+    # -------------------------- INSERT steps functions -----------------------
+    def _insertAllSteps(self):
+
+        for tomo in self.inTomograms.get():
+
+            self._insertFunctionStep(self.runTomoResample,
+                        tomo.getFileName())
+
+        self._insertFunctionStep(self.createOutputStep)
+
+    def runTomoResample(self, tomoFile: str):
+
+        prog = Plugin.getProgram('resample')
+
+
+        tomoBaseName = removeBaseExt(tomoFile)
+        tomoExt = getExt(tomoFile)
+        paramDict = {}
+        paramDict['tomoFile'] = tomoFile
+        paramDict['tomoOutName'] = self.getWorkingDir() + '/' + OUTPUT_DIR + '/' + \
```

Comment:
this whole thing can be replaced with self._getExtraPath(tomoBaseName + '_resampled' + tomoExt)